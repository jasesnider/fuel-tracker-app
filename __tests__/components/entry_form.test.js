import React from 'react';
import renderer from 'react-test-renderer';
import EntryForm from '../../src/components/entry_form';

describe('EntryForm', () => {
  it('renders correctly', () => {
    const props = {
      currency: '$',
      saveEntry: () => {},
      fuelEconomy: 'MPG',
      cost: '34.23',
      quantity: '12',
      odometer: '1321321',
      isFullTank: true,
      onChangeHandler: () => {},
    };

    const tree = renderer
      .create(
        <EntryForm
          currency={props.currency}
          cost={props.cost}
          price={props.cost / props.quantity}
          quantity={props.quantity}
          isFullTank={props.isFullTank}
          odometer={props.odometer}
          fuelEconomy={props.fuelEconomy}
          saveEntry={props.saveEntry}
          onChangeHandler={props.onChangeHandler}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
