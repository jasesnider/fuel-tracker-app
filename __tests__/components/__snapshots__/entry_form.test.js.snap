// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EntryForm renders correctly 1`] = `
<div
  className="form-wrapper"
>
  <form>
    <div
      className="input-field-wrapper"
    >
      Fuel Economy: MPG
    </div>
    <div
      className="input-field-wrapper"
    >
      <span>
        Cost: 
      </span>
      <div
        className="ant-input-number"
        style={Object {}}
      >
        <div
          className="ant-input-number-handler-wrap"
        >
          <span
            aria-disabled={false}
            aria-label="Increase Value"
            className="ant-input-number-handler ant-input-number-handler-up "
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchCancel={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            unselectable="unselectable"
          >
            <i
              className="anticon anticon-up ant-input-number-handler-up-inner"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="up"
                fill="currentColor"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M890.5 755.3L537.9 269.2c-12.8-17.6-39-17.6-51.7 0L133.5 755.3A8 8 0 0 0 140 768h75c5.1 0 9.9-2.5 12.9-6.6L512 369.8l284.1 391.6c3 4.1 7.8 6.6 12.9 6.6h75c6.5 0 10.3-7.4 6.5-12.7z"
                />
              </svg>
            </i>
          </span>
          <span
            aria-disabled={false}
            aria-label="Decrease Value"
            className="ant-input-number-handler ant-input-number-handler-down "
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchCancel={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            unselectable="unselectable"
          >
            <i
              className="anticon anticon-down ant-input-number-handler-down-inner"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="down"
                fill="currentColor"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                />
              </svg>
            </i>
          </span>
        </div>
        <div
          aria-valuemin={0}
          aria-valuenow={34.23}
          className="ant-input-number-input-wrap"
          role="spinbutton"
        >
          <input
            autoComplete="off"
            className="ant-input-number-input"
            min={0}
            name="cost"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseUp={[Function]}
            required={false}
            step={1}
            value="34.23"
          />
        </div>
      </div>
    </div>
    <div
      className="input-field-wrapper"
    >
      <span>
        Quantity: 
      </span>
      <div
        className="ant-input-number"
        style={Object {}}
      >
        <div
          className="ant-input-number-handler-wrap"
        >
          <span
            aria-disabled={false}
            aria-label="Increase Value"
            className="ant-input-number-handler ant-input-number-handler-up "
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchCancel={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            unselectable="unselectable"
          >
            <i
              className="anticon anticon-up ant-input-number-handler-up-inner"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="up"
                fill="currentColor"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M890.5 755.3L537.9 269.2c-12.8-17.6-39-17.6-51.7 0L133.5 755.3A8 8 0 0 0 140 768h75c5.1 0 9.9-2.5 12.9-6.6L512 369.8l284.1 391.6c3 4.1 7.8 6.6 12.9 6.6h75c6.5 0 10.3-7.4 6.5-12.7z"
                />
              </svg>
            </i>
          </span>
          <span
            aria-disabled={false}
            aria-label="Decrease Value"
            className="ant-input-number-handler ant-input-number-handler-down "
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchCancel={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            unselectable="unselectable"
          >
            <i
              className="anticon anticon-down ant-input-number-handler-down-inner"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="down"
                fill="currentColor"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                />
              </svg>
            </i>
          </span>
        </div>
        <div
          aria-valuemin={0}
          aria-valuenow={12}
          className="ant-input-number-input-wrap"
          role="spinbutton"
        >
          <input
            autoComplete="off"
            className="ant-input-number-input"
            min={0}
            name="quantity"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseUp={[Function]}
            required={false}
            step={1}
            value="12"
          />
        </div>
      </div>
    </div>
    <div
      className="input-field-wrapper"
    >
      <span>
        Odometer: 
      </span>
      <div
        className="ant-input-number"
        style={Object {}}
      >
        <div
          className="ant-input-number-handler-wrap"
        >
          <span
            aria-disabled={false}
            aria-label="Increase Value"
            className="ant-input-number-handler ant-input-number-handler-up "
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchCancel={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            unselectable="unselectable"
          >
            <i
              className="anticon anticon-up ant-input-number-handler-up-inner"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="up"
                fill="currentColor"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M890.5 755.3L537.9 269.2c-12.8-17.6-39-17.6-51.7 0L133.5 755.3A8 8 0 0 0 140 768h75c5.1 0 9.9-2.5 12.9-6.6L512 369.8l284.1 391.6c3 4.1 7.8 6.6 12.9 6.6h75c6.5 0 10.3-7.4 6.5-12.7z"
                />
              </svg>
            </i>
          </span>
          <span
            aria-disabled={false}
            aria-label="Decrease Value"
            className="ant-input-number-handler ant-input-number-handler-down "
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchCancel={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            unselectable="unselectable"
          >
            <i
              className="anticon anticon-down ant-input-number-handler-down-inner"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="down"
                fill="currentColor"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                />
              </svg>
            </i>
          </span>
        </div>
        <div
          aria-valuemin={0}
          aria-valuenow={1321321}
          className="ant-input-number-input-wrap"
          role="spinbutton"
        >
          <input
            autoComplete="off"
            className="ant-input-number-input"
            min={0}
            name="odometer"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseUp={[Function]}
            required={false}
            step={1}
            value="1321321"
          />
        </div>
      </div>
    </div>
    <div
      className="input-field-wrapper"
    >
       
      <label>
        Full Tank?
        <label
          className="ant-checkbox-wrapper ant-checkbox-wrapper-checked"
        >
          <span
            className="ant-checkbox ant-checkbox-checked"
            style={Object {}}
          >
            <input
              checked={true}
              className="ant-checkbox-input"
              name="isFullTank"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              type="checkbox"
              value={true}
            />
            <span
              className="ant-checkbox-inner"
            />
          </span>
        </label>
      </label>
    </div>
    <div
      className="input-field-wrapper"
    >
       
      <label>
        Price Per Gallon ($)
        <div
          className="ant-input-number"
          style={Object {}}
        >
          <div
            className="ant-input-number-handler-wrap"
          >
            <span
              aria-disabled={true}
              aria-label="Increase Value"
              className="ant-input-number-handler ant-input-number-handler-up "
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              role="button"
              unselectable="unselectable"
            >
              <i
                className="anticon anticon-up ant-input-number-handler-up-inner"
              >
                <svg
                  aria-hidden="true"
                  className=""
                  data-icon="up"
                  fill="currentColor"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M890.5 755.3L537.9 269.2c-12.8-17.6-39-17.6-51.7 0L133.5 755.3A8 8 0 0 0 140 768h75c5.1 0 9.9-2.5 12.9-6.6L512 369.8l284.1 391.6c3 4.1 7.8 6.6 12.9 6.6h75c6.5 0 10.3-7.4 6.5-12.7z"
                  />
                </svg>
              </i>
            </span>
            <span
              aria-disabled={true}
              aria-label="Decrease Value"
              className="ant-input-number-handler ant-input-number-handler-down "
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              role="button"
              unselectable="unselectable"
            >
              <i
                className="anticon anticon-down ant-input-number-handler-down-inner"
              >
                <svg
                  aria-hidden="true"
                  className=""
                  data-icon="down"
                  fill="currentColor"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                  />
                </svg>
              </i>
            </span>
          </div>
          <div
            aria-valuemin={-9007199254740991}
            aria-valuenow={2.8524999999999996}
            className="ant-input-number-input-wrap"
            role="spinbutton"
          >
            <input
              autoComplete="off"
              className="ant-input-number-input"
              min={-9007199254740991}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseUp={[Function]}
              readOnly={true}
              required={false}
              step={1}
              value="2.8524999999999996"
            />
          </div>
        </div>
      </label>
    </div>
  </form>
  <style
    jsx={true}
  >
    
          .form-wrapper {
            background-color: white;
            margin: 0 auto;
            height: auto;
            padding: 10px;
            z-index: 1000;
            }
          }

          .form-wrapper input {
            margin: 10px auto;
            padding: 4px;
          }
          .entry-submit {
            border-radius: 4px;
            background-color: #17a2b8;
            color: white;
            padding: 3px 6px;
          }
          .input-field-wrapper {
            margin-bottom: 15px;
          }
        
  </style>
</div>
`;
